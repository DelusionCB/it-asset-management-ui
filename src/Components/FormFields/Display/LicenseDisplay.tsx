import {Container} from 'reactstrap';
import React from 'react';
import {LicenseDataProps} from '../../../Types/types.directories';
import {DisplayDescription, DisplayHeader, DisplayText} from '../../CustomComponents/DisplayComponents';
import './index.scss';

interface licDisplayProps {
    values: LicenseDataProps
    t: any
    navigate: any
}

function LicenseDisplay ({values, t, navigate}: licDisplayProps): JSX.Element {
    return (
        <Container className='directory-wrapper'>

            <DisplayHeader t={t} label='license.lcn' />

            <DisplayDescription t={t} label='license.description' />

            <DisplayText t={t} value={values.name} label='values.name' />
            <DisplayText t={t} value={values.description} label='values.description' />

            <DisplayText t={t} value={values.valid_from_date} label='values.valid_from_date' />
            <DisplayText t={t} value={values.valid_until_date} label='values.valid_until_date' />

            <DisplayText t={t} value={values.license_type} label='values.license_type' />

            <DisplayText t={t} value={values.contract} label='values.contract' />
            <DisplayText t={t} value={values.fileUrl} label='values.fileUrl' />

        </Container>
    )
}

export default LicenseDisplay;
