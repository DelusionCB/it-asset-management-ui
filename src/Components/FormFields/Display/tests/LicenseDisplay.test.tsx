import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import LicenseDisplay from '../LicenseDisplay'
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import {licenseDefaultValues} from '../../../../Types/types.directories';

describe('<LicenseDisplay />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <LicenseDisplay t={jest.fn()} navigate={jest.fn()} values={licenseDefaultValues} />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
