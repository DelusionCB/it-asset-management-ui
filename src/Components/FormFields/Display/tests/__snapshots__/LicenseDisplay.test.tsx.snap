// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LicenseDisplay /> Snapshot renders correctly 1`] = `
<div
  className="directory-wrapper container"
>
  <div
    className="display-header row"
  >
    <h1 />
  </div>
  <div
    className="display-description row"
  >
    <h2 />
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <div>
      <h4 />
    </div>
  </div>
  <div
    className="action-buttons"
  >
    <button
      className="action-button btn btn-secondary disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      button.copy
    </button>
    <button
      className="action-button btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      button.edit
    </button>
  </div>
</div>
`;
